src/frontend/overlay/src/components/organisms/holder/TimeLine.jsx

```diff
@@ -0,0 +1,184 @@
+    border-top-left-radius: ${c.TIMELINE_BORDER_RADIUS};
+    border-bottom-right-radius: ${c.TIMELINE_BORDER_RADIUS};;
+    display: grid;
+    height: ${c.TIMELINE_WRAP_HEIGHT};
+    background-color: ${c.CONTEST_COLOR};
+`;
+
+
+const Line = styled.div`
+    width: 100%;
+    height: ${c.TIMLINE_LINE_HEIGHT};
+    background: linear-gradient(270deg, #D13D23 -28.28%, #FFC239 33.33%, #1A63D8 100%);
+    position: relative;
+`;
+
+const Circle = styled.div`
+    width: ${c.TIMLINE_CIRCLE_RADIUS};
+    height: ${c.TIMLINE_CIRCLE_RADIUS};
+    background-color: ${( { color } ) => color };
+    border-radius: 50%;
+    position: absolute;
+    align-content: center;
+`;
+
+const Label = styled.div`
+    position: relative;
+    justify-content: center;
+    display: flex;
+    align-self: center;
+    text-align: center;
+    align-items: center;
+    color: ${({ darkText }) => darkText ? "#000" : "#FFF"};
+    font-weight: ${({ isBold }) => isBold ? "bold" : "normal"};
+`;
+
+const ProblemWrap = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    position: absolute;
+    left: ${({ left }) => left};
+`;
+
+const TextWithAnimation = styled.div`
+    animation: ${ChangeTextAnimation} 10s infinite;
+    justify-content: center;
+    position: absolute;
+    align-items: center;
+    text-align: center;
+`;
+
+const TextWithAnimation2 = styled.div`
+    animation: ${ChangeTextAnimation2} 10s infinite;
+    justify-content: center;
+    position: absolute;
+    align-items: center;
+    text-align: center;
+    font-size: 12px;
+`;
+
+const Text = styled.div`
+    position: absolute;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    text-align: center;
+`;
+
+
+export const TimeLine = ({ className, teamId }) => {
+    const contestInfo = useAppSelector(state => state.contestInfo.info);
+    const [runsResults, setRunsResults] = useState([]);
+    
+    const getColor = (problemResult) => {
+        if (problemResult.type === "IN_PROGRESS") {
+            return c.VERDICT_UNKNOWN;
+        } else if (problemResult.type === "ICPC") {
+            if (problemResult.isAccepted) {
+                return c.VERDICT_OK;
+            } else {
+                return c.VERDICT_NOK;
+            }
+        } else {
+            const task = contestInfo.problems.find(info => info.letter === problemResult.problemId);
+            return getIOIColor(problemResult.score, task?.minScore, task?.maxScore);
+        }
+    };
+
+    const Problem = ({ problemResult, color, contestLengthMs }) => {
+        let left = (100 * problemResult.time / contestLengthMs) * 0.99;
+        const darkText = isShouldUseDarkColor(color);
+        return (
+            <ProblemWrap left={left + "%"}>
+                <Circle color={getColor(problemResult)}>
+                    <Label darkText={darkText} isBold={problemResult.type === "IN_PROGRESS"}>
```

Comment:
Let's use generated ts constants instead. You can add class into list in build.gradle.kts ins scheme generator, if it is not generated.